### PR TITLE
[Doc] Update Node client document

### PR DESCRIPTION
**Pull request by cimura**
### Motivation

Update the document according to Node client 1.3.0.

### Modifications

- Add [new features of Node.js client 1.3.0](https://github.com/apache/pulsar-client-node/releases/tag/v1.3.0) to the document.
- Fix some grammatical errors.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: site2/docs/client-libraries-node.md [hunk 1]
```diff
@@ -125,9 +128,11 @@ Pulsar Node.js producers have the following methods available:
 
 | Method | Description | Return type |
 | :----- | :---------- | :---------- |
-| `send(Object)` | Publishes a [message](#messages) to the producer's topic. When the message is successfully acknowledged by the Pulsar broker, or an error will be thrown, the Promise object run executor function. | `Promise<null>` |
-| `flush()` | Sends message from send queue to Pulser broker. When the message is successfully acknowledged by the Pulsar broker, or an error will be thrown, the Promise object run executor function. | `Promise<null>` |
-| `close()` | Closes the producer and releases all resources allocated to it. If `close()` is called then no more messages will be accepted from the publisher. This method will return Promise object, and when all pending publish requests have been persisted by Pulsar then run executor function. If an error is thrown, no pending writes will be retried. | `Promise<null>` |
+| `send(Object)` | Publishes a [message](#messages) to the producer's topic. When the message is successfully acknowledged by the Pulsar broker, or an error will be thrown, the Promise object whose result is the message ID runs executor function. | `Promise<Object>` |
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-node.md:131]:
```suggestion
| `send(Object)` | Publishes a [message](#messages) to the producer's topic. When the message is successfully acknowledged by the Pulsar broker, or an error will be thrown, the Promise object whose result is the message ID runs executor function. | `Promise<Object>` |  
```
For the latter sentence, do you mean "When the message is successfully acknowledged by the Pulsar broker or an error is thrown, the Promise object whose result is the message ID runs executor function. 

Use present tense if you are covering facts that were, are, and forever shall be true. Always use the present tense in technical writing by default.

File: site2/docs/client-libraries-node.md [hunk 2]
```diff
@@ -125,9 +128,11 @@ Pulsar Node.js producers have the following methods available:
...
+| `flush()` | Sends message from send queue to Pulser broker. When the message is successfully acknowledged by the Pulsar broker, or an error will be thrown, the Promise object runs executor function. | `Promise<null>` |
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-node.md:132]: ditto

File: site2/docs/client-libraries-node.md [hunk 3]
```diff
@@ -125,9 +128,11 @@ Pulsar Node.js producers have the following methods available:
...
+| `close()` | Closes the producer and releases all resources allocated to it. If `close()` is called then no more messages will be accepted from the publisher. This method will return Promise object, and when all pending publish requests have been persisted by Pulsar then runs executor function. If an error is thrown, no pending writes will be retried. | `Promise<null>` |
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-node.md:133]:
```suggestion
| `close()` | Closes the producer and releases all resources allocated to it. If `close()` is called then no more messages will be accepted from the publisher. This method will return Promise object, and when all pending publish requests have been persisted by Pulsar then runs executor function. If an error is thrown, no pending writes will be retried. | `Promise<null>` |
```

1. Use present tense if you are covering facts that were, are, and forever shall be true. Always use the present tense in technical writing by default.

2. Shorten long sentences.

File: site2/docs/client-libraries-node.md [hunk 4]
```diff
@@ -125,9 +128,11 @@ Pulsar Node.js producers have the following methods available:
...
+| `getProducerName()` | Getter method of the name of the producer. | `string` |
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-node.md:134]: Do you mean " Getter method of the producer name"?

File: site2/docs/client-libraries-node.md [hunk 5]
```diff
@@ -389,6 +397,8 @@ The following keys are available for producer message objects:
 | `sequenceId` | The sequence ID of the message. |
 | `partitionKey` | The optional key associated with the message (particularly useful for things like topic compaction). |
 | `replicationClusters` | The clusters to which this message will be replicated. Pulsar brokers handle message replication automatically; you should only change this setting if you want to override the broker default. |
+| `deliverAt` | The absolute timestamp at or after which the message is delivered. | |
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-node.md:400]: Include "after"?

File: site2/docs/client-libraries-node.md (hunk 3 above)
Comment (cimura) [reply site2/docs/client-libraries-node.md]: fixed this. https://github.com/apache/pulsar/pull/10045/commits/7a6356373ea03a11948c474f5616592148c48bf1

File: site2/docs/client-libraries-node.md (hunk 1 above)
Comment (cimura) [reply site2/docs/client-libraries-node.md]: fixed this. https://github.com/apache/pulsar/pull/10045/commits/70bcc41a5d79e5b2f19644eb9fd0aeed765234f8

File: site2/docs/client-libraries-node.md (hunk 5 above)
Comment (cimura) [reply site2/docs/client-libraries-node.md:400]: I thought so
reference: 

- https://github.com/apache/pulsar/blob/3b2c8526a96fc90e948dbeb510e8400a628c749a/pulsar-client-api/src/main/java/org/apache/pulsar/client/api/TypedMessageBuilder.java#L187
- https://github.com/apache/pulsar/blame/30f3ddab9e1bf36f8361c3e58a468cda9a9c0b99/site2/docs/client-libraries-go.md#L671

File: site2/docs/client-libraries-node.md [hunk 6]
```diff
@@ -219,12 +224,15 @@ Pulsar Node.js consumers have the following methods available:
...
 | `negativeAcknowledgeId(MessageId)` | [Negatively acknowledges](reference-terminology.md#negative-acknowledgment-nack) a message to the Pulsar broker by message ID object. | `void` |
 | `close()` | Closes the consumer, disabling its ability to receive messages from the broker. | `Promise<null>` |
+| `unsubscribe()` | Unsubscribes the subscription. | `Promise<null>` |
 
 ### Consumer configuration
 
 | Parameter | Description | Default |
 | :-------- | :---------- | :------ |
-| `topic` | The Pulsar [topic](reference-terminology.md#topic) on which the consumer will establish a subscription and listen for messages. | |
+| `topic` | The Pulsar topic on which the consumer will establish a subscription and listen for messages. | |
```
Comment (Anonymitaet) [inline site2/docs/client-libraries-node.md:233]: same comment as previously (use present tense)

File: site2/docs/client-libraries-node.md (hunk 5 above)
Comment (Anonymitaet) [reply site2/docs/client-libraries-node.md:400]: OK, thanks

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**